Fix test failures on TestRunResourceServiceBlackBoxTest

See https://ci.centos.org/view/Devtools/job/devtools-fabric8-auth-build-master/326/console:

```
--- FAIL: TestRunResourceServiceBlackBoxTest (0.49s)
    --- FAIL: TestRunResourceServiceBlackBoxTest/TestRegisterReadDeleteResourceWithParentOK (0.03s)
    	assertions.go:247:                           
	Error Trace:	resource_service_blackbox_test.go:101    		
	Error:      	Not equal:     		
	            	expected: []string{"view", "contribute", "manage"}    		
	            	actual  : []string{"view", "contribute", "manage", "space-scope1", "space-scope2", "space-scope3"}
```

Apparently, so left-over data is causing errors on this test.

--- a/authorization/resource/repository/resource_blackbox_test.go
+++ b/authorization/resource/repository/resource_blackbox_test.go
@@ -200,106 +200,109 @@ func createAndLoadResource(s *resourceBlackBoxTest, parentResourceID *string) *r
 
 func (s *resourceBlackBoxTest) TestFindWithRoleByResourceTypeAndIdentity() {
 
-	spaceType := s.Graph.LoadResourceType(authorization.ResourceTypeSpace)
-	organizationType := s.Graph.LoadResourceType(authorization.IdentityResourceTypeOrganization)
-
 	s.T().Run("individual", func(t *testing.T) {
 
-		t.Run("individual is admin on no space", func(t *testing.T) {
-			// given
-			g := s.NewTestGraph(s.T())
-			user := g.CreateUser()
-			space := g.CreateSpace()
-			require.Equal(t, authorization.ResourceTypeSpace, space.Resource().ResourceType.Name)
-			// when
-			resources, err := s.repo.FindWithRoleByResourceTypeAndIdentity(
-				context.Background(),
-				authorization.ResourceTypeSpace,
-				user.IdentityID())
-			// then
-			require.NoError(t, err)
-			assert.Len(t, resources, 0)
-		})
-
-		t.Run("individual is admin on 1 space", func(t *testing.T) {
-			// given
-			g := s.NewTestGraph(t)
-			role := g.CreateRole(spaceType).AddScope("space-scope1").AddScope("space-scope2").AddScope("space-scope3")
-			user := g.CreateUser()
-			space := g.CreateSpace().AddRole(user, role)
-			require.Equal(t, authorization.ResourceTypeSpace, space.Resource().ResourceType.Name)
-			// when
-			resources, err := s.repo.FindWithRoleByResourceTypeAndIdentity(
-				context.Background(),
-				authorization.ResourceTypeSpace,
-				user.IdentityID())
-			// then
-			require.NoError(t, err)
-			assert.Len(t, resources, 1)
-			assert.Equal(t, space.Resource().ResourceID, resources[0])
-		})
-
-		t.Run("individual is admin on 2 spaces", func(t *testing.T) {
-			// given
-			g := s.NewTestGraph(t)
-			role := g.CreateRole(spaceType).AddScope("space-scope1").AddScope("space-scope2").AddScope("space-scope3")
-			user := g.CreateUser()
-			space1 := g.CreateSpace().AddRole(user, role)
-			require.Equal(t, authorization.ResourceTypeSpace, space1.Resource().ResourceType.Name)
-			space2 := g.CreateSpace().AddRole(user, role)
-			require.Equal(t, authorization.ResourceTypeSpace, space2.Resource().ResourceType.Name)
-			g.CreateSpace() // another space on which the user has no role
-			// when
-			resources, err := s.repo.FindWithRoleByResourceTypeAndIdentity(
-				context.Background(),
-				authorization.ResourceTypeSpace,
-				user.IdentityID())
-			// then
-			require.NoError(t, err)
-			require.Len(t, resources, 2)
-			// resources should be space1 and space2, not space3
-			assert.ElementsMatch(t, []string{space1.SpaceID(), space2.SpaceID()}, resources)
-		})
-
-		t.Run("individual is admin in the parent organization but no default or custom role mapping", func(t *testing.T) {
+		t.Run("individual is admin on no resource", func(t *testing.T) {
 			// given
 			g := s.NewTestGraph(t)
 			user := g.CreateUser()
-			org := g.CreateOrganization(user) // user will be creator and have a role in the org
-			g.CreateSpace(org)
-			// here we don't map the admin role in the org to a contributor role in the space,
-			// so the user is not considered as a contributor in the created space
+			rt := g.CreateResourceType()
+			r := g.CreateResource(rt)
+			require.Equal(t, rt.ResourceType().Name, r.Resource().ResourceType.Name)
 			// when
 			resources, err := s.repo.FindWithRoleByResourceTypeAndIdentity(
 				context.Background(),
-				authorization.ResourceTypeSpace,
+				rt.ResourceType().Name,
 				user.IdentityID())
 			// then
 			require.NoError(t, err)
 			assert.Empty(t, resources)
 		})
 
-		t.Run("individual is contributor in the parent organization with default role mapping", func(t *testing.T) {
+		t.Run("individual is admin on 1 resource", func(t *testing.T) {
 			// given
 			g := s.NewTestGraph(t)
-			orgRole := g.CreateRole(organizationType).AddScope("org-scope1").AddScope("org-scope2").AddScope("org-scope3")
-			spaceRole := g.CreateRole(spaceType).AddScope("space-scope1").AddScope("space-scope2").AddScope("space-scope3")
+			rt := g.CreateResourceType()
+			role := g.CreateRole(rt).AddScope("resource-scope1").AddScope("space-scope2").AddScope("space-scope3")
 			user := g.CreateUser()
-			org := g.CreateOrganization(user).AddRole(user, orgRole) // user will be creator and have a role in the org
-			space := g.CreateSpace(org)
-			// here we map the role in the org to another role in the space,
-			// so the user also inherits a role in the created space
-			spaceType := g.ResourceTypeByID(space.Resource().ResourceType.ResourceTypeID)
-			g.CreateDefaultRoleMapping(spaceType, orgRole, spaceRole)
+			r := g.CreateResource(rt).AddRole(user, role)
+			require.Equal(t, rt.Name(), r.Resource().ResourceType.Name)
 			// when
 			resources, err := s.repo.FindWithRoleByResourceTypeAndIdentity(
 				context.Background(),
-				authorization.ResourceTypeSpace,
+				rt.Name(),
+				user.IdentityID())
+			// then
+			require.NoError(t, err)
+			assert.Len(t, resources, 1)
+			assert.Equal(t, r.Resource().ResourceID, resources[0])
+		})
+
+		t.Run("individual is admin on 2 resources", func(t *testing.T) {
+			// given
+			g := s.NewTestGraph(t)
+			rt := g.CreateResourceType()
+			role := g.CreateRole(rt).AddScope("space-scope1").AddScope("space-scope2").AddScope("space-scope3")
+			user := g.CreateUser()
+			r1 := g.CreateResource(rt).AddRole(user, role)
+			require.Equal(t, rt.Name(), r1.Resource().ResourceType.Name)
+			r2 := g.CreateResource(rt).AddRole(user, role)
+			require.Equal(t, rt.Name(), r2.Resource().ResourceType.Name)
+			g.CreateSpace() // another space on which the user has no role
+			// when
+			resources, err := s.repo.FindWithRoleByResourceTypeAndIdentity(
+				context.Background(),
+				rt.Name(),
+				user.IdentityID())
+			// then
+			require.NoError(t, err)
+			require.Len(t, resources, 2)
+			// resources should be space1 and space2, not space3
+			assert.ElementsMatch(t, []string{r1.ResourceID(), r2.ResourceID()}, resources)
+		})
+
+		t.Run("individual is admin in the parent resource but no default or custom role mapping", func(t *testing.T) {
+			// given
+			g := s.NewTestGraph(t)
+			user := g.CreateUser()
+			parentResourceType := g.CreateResourceType()
+			parentResource := g.CreateResource(user, parentResourceType) // user will be creator and have a role in the org
+			childResourceType := g.CreateResourceType()
+			g.CreateResource(childResourceType, parentResource)
+			// here we don't map the admin role in the org to a contributor role in the space,
+			// so the user is not considered as a contributor in the created space
+			// when
+			resources, err := s.repo.FindWithRoleByResourceTypeAndIdentity(
+				context.Background(),
+				childResourceType.Name(),
+				user.IdentityID())
+			// then
+			require.NoError(t, err)
+			assert.Empty(t, resources)
+		})
+
+		t.Run("individual is contributor in the parent resource with default role mapping", func(t *testing.T) {
+			// given
+			g := s.NewTestGraph(t)
+			user := g.CreateUser()
+			parentType := g.CreateResourceType()
+			parentRole := g.CreateRole(parentType).AddScope("parent-scope1").AddScope("parent-scope2").AddScope("parent-scope3")
+			parentResource := g.CreateResource(parentType).AddRole(user, parentRole) // user will be creator and have a role in the org
+			childType := g.CreateResourceType()
+			childResource := g.CreateResource(childType, parentResource)
+			// here we map the role in the org to another role in the space,
+			// so the user also inherits a role in the created space
+			childRole := g.CreateRole(childType).AddScope("child-scope1").AddScope("child-scope2").AddScope("child-scope3")
+			g.CreateDefaultRoleMapping(childType, parentRole, childRole)
+			// when
+			resources, err := s.repo.FindWithRoleByResourceTypeAndIdentity(
+				context.Background(),
+				childType.Name(),
 				user.IdentityID())
 			// then
 			require.NoError(t, err)
 			require.Len(t, resources, 1)
-			assert.Equal(t, space.Resource().ResourceID, resources[0])
+			assert.Equal(t, childResource.Resource().ResourceID, resources[0])
 		})
 
 		t.Run("individual is admin in the parent organization with custom role mapping", func(t *testing.T) {
@@ -310,66 +313,68 @@ func (s *resourceBlackBoxTest) TestFindWithRoleByResourceTypeAndIdentity() {
 
 	s.T().Run("teams", func(t *testing.T) {
 
-		t.Run("individual belongs to admin team on no space", func(t *testing.T) {
+		t.Run("individual belongs to admin team on no resource", func(t *testing.T) {
 			// given
 			g := s.NewTestGraph(t)
 			user := g.CreateUser()
 			g.CreateTeam("team").AddMember(user)
-			space := g.CreateSpace()
-			require.Equal(t, authorization.ResourceTypeSpace, space.Resource().ResourceType.Name)
+			rt := g.CreateResourceType()
+			require.Equal(t, rt.Name(), rt.Name())
 			// when
 			resources, err := s.repo.FindWithRoleByResourceTypeAndIdentity(
 				context.Background(),
-				authorization.ResourceTypeSpace,
+				rt.Name(),
 				user.IdentityID())
 			// then
 			require.NoError(t, err)
 			assert.Len(t, resources, 0)
 		})
 
-		t.Run("individual belongs to admin team on 1 space", func(t *testing.T) {
+		t.Run("individual belongs to admin team on 1 resource", func(t *testing.T) {
 			// given
 			g := s.NewTestGraph(t)
 			user := g.CreateUser()
 			team := g.CreateTeam("team").AddMember(user)
-			role := g.CreateRole(spaceType).AddScope("org-scope1").AddScope("org-scope2").AddScope("org-scope3")
-			space := g.CreateSpace().AddRole(team, role)
-			require.Equal(t, authorization.ResourceTypeSpace, space.Resource().ResourceType.Name)
+			rt := g.CreateResourceType()
+			role := g.CreateRole(rt).AddScope("resource-scope1").AddScope("resource-scope2").AddScope("resource-scope3")
+			r := g.CreateResource(rt).AddRole(team, role)
+			require.Equal(t, rt.Name(), rt.Name())
 			// when
 			resources, err := s.repo.FindWithRoleByResourceTypeAndIdentity(
 				context.Background(),
-				authorization.ResourceTypeSpace,
+				rt.Name(),
 				user.IdentityID())
 			// then
 			require.NoError(t, err)
 			require.Len(t, resources, 1)
-			assert.Equal(t, space.Resource().ResourceID, resources[0])
+			assert.Equal(t, r.Resource().ResourceID, resources[0])
 		})
 
-		t.Run("individual is member of admin team in the parent organization with default role mapping", func(t *testing.T) {
+		t.Run("individual is member of admin team in the parent resource with default role mapping", func(t *testing.T) {
 			// given
 			g := s.NewTestGraph(t)
 			creator := g.CreateUser()
-			org := g.CreateOrganization(creator) // team (hence user) will be creator and admin of the org
+			parentResourceType := g.CreateResourceType()                    // team (hence user) will be creator and admin of the org
+			parentResource := g.CreateResource(creator, parentResourceType) // team (hence user) will be creator and admin of the org
 			user := g.CreateUser()
 			team := g.CreateTeam("team").AddMember(user)
-			orgRole := g.CreateRole(organizationType).AddScope("org-scope1").AddScope("org-scope2").AddScope("org-scope3")
-			org.AddRole(team, orgRole)
-			space := g.CreateSpace(org)
+			parentRole := g.CreateRole(parentResourceType).AddScope("parent-scope1").AddScope("parent-scope2").AddScope("parent-scope3")
+			parentResource.AddRole(team, parentRole)
+			childResourceType := g.CreateResourceType()
+			childResource := g.CreateResource(childResourceType, parentResource)
 			// here we map the admin role in the org to a contributor role in the space,
 			// so the user is also considered as a contributor in the created space
-			spaceRole := g.CreateRole(organizationType).AddScope("space-scope1").AddScope("space-scope2").AddScope("space-scope3")
-			spaceType := g.ResourceTypeByID(space.Resource().ResourceType.ResourceTypeID)
-			g.CreateDefaultRoleMapping(spaceType, orgRole, spaceRole)
+			childRole := g.CreateRole(childResourceType).AddScope("child-scope1").AddScope("child-scope2").AddScope("child-scope3")
+			g.CreateDefaultRoleMapping(childResourceType, parentRole, childRole)
 			// when
 			resources, err := s.repo.FindWithRoleByResourceTypeAndIdentity(
 				context.Background(),
-				authorization.ResourceTypeSpace,
+				childResourceType.Name(),
 				user.IdentityID())
 			// then
 			require.NoError(t, err)
 			require.Len(t, resources, 1)
-			assert.Equal(t, space.Resource().ResourceID, resources[0])
+			assert.Equal(t, childResource.Resource().ResourceID, resources[0])
 		})
 
 		t.Run("individual is member of admin team in the parent organization with custom role mapping", func(t *testing.T) {

--- a/test/graph/resource_type_wrapper.go
+++ b/test/graph/resource_type_wrapper.go
@@ -60,6 +60,10 @@ func (w *resourceTypeWrapper) ResourceType() *resourcetype.ResourceType {
 	return w.resourceType
 }
 
+func (w *resourceTypeWrapper) Name() string {
+	return w.resourceType.Name
+}
+
 func (w *resourceTypeWrapper) AddScope(scope string) {
 	rts := &resourcetype.ResourceTypeScope{
 		ResourceTypeID: w.resourceType.ResourceTypeID,

--- a/test/graph/resource_wrapper.go
+++ b/test/graph/resource_wrapper.go
@@ -53,6 +53,7 @@ func newResourceWrapper(g *TestGraph, params []interface{}) interface{} {
 
 	if resourceType == nil {
 		resourceType = w.graph.CreateResourceType().ResourceType()
+		g.t.Logf("created new resource type with name='%v' (%v)", resourceType.Name, resourceType.ResourceTypeID)
 	}
 
 	if resourceName == nil {

